core/templates/dev/head/pages/creator_dashboard/CreatorDashboard.js

```diff
@@ -319,196 +321,15 @@ oppia.controller('CreatorDashboard', [
-            };
-            $scope.isResubmitButtonShown = function() {
-              return (
-                !$scope.isNotHandled && $scope.isSuggestionRejected &&
-                $scope.suggestionEditorIsShown);
-            };
-            $scope.isResubmitButtonDisabled = function() {
-              return !(
-                $scope.summaryMessage &&
-                ($scope.suggestionData.newSuggestionHtml.trim() !==
-                  newContent.html.trim()));
-            };
-            $scope.cancelEditMode = function() {
-              $scope.suggestionEditorIsShown = false;
-            };
-            $scope.resubmitChanges = function() {
-              $uibModalInstance.close({
-                action: ACTION_RESUBMIT_SUGGESTION,
-                newSuggestionHtml: $scope.suggestionData.newSuggestionHtml,
-                summaryMessage: $scope.summaryMessage,
-                stateName: $scope.stateName,
-                suggestionType: $scope.suggestionType,
-                oldContent: $scope.oldContent
-              });
-            };
-          }
-        ]
-      }).result.then(function(result) {
-        var RESUBMIT_SUGGESTION_URL_TEMPLATE = (
-          '/suggestionactionhandler/resubmit/<suggestion_id>');
-        var HANDLE_SUGGESTION_URL_TEMPLATE = (
-          '/suggestionactionhandler/<target_type>/<target_id>/<suggestion_id>');
-
-        var url = null;
-        var data = null;
-        if (result.action === 'resubmit' &&
-            result.suggestionType === 'edit_exploration_state_content') {
-          url = UrlInterpolationService.interpolateUrl(
-            RESUBMIT_SUGGESTION_URL_TEMPLATE, {
-              suggestion_id: $scope.activeThread.suggestion.suggestionId
-            }
-          );
-          data = {
-            action: result.action,
-            summary_message: result.summaryMessage,
-            change: {
-              cmd: 'edit_state_property',
-              property_name: 'content',
-              state_name: result.stateName,
-              old_value: result.oldContent,
-              new_value: {
-                html: result.newSuggestionHtml
-              }
-            }
-          };
-        } else {
-          url = UrlInterpolationService.interpolateUrl(
-            HANDLE_SUGGESTION_URL_TEMPLATE, {
-              target_type: $scope.activeThread.suggestion.targetType,
-              target_id: $scope.activeThread.suggestion.targetId,
-              suggestion_id: $scope.activeThread.suggestion.suggestionId
-            }
-          );
-          data = {
-            action: result.action,
-            commit_message: result.commitMessage,
-            review_message: result.reviewMessage
-          };
+      ShowSuggestionModalForCreatorViewService.showSuggestionModal(
+        $scope.activeThread.suggestion.suggestionType,
+        {
+          activeThread: $scope.activeThread,
+          suggestionsToReviewList: $scope.suggestionsToReviewList,
```

Comment:
Why do we need to pass this in?

Reply:
The parameters passed here are required to handle a suggestion in the SuggestionService for creator view (https://github.com/oppia/oppia/pull/6093/files#diff-3cd0164785e4bfe2277b36148961a0c2R30). So, we need to pass them.